omegaconf/basecontainer.py

```diff
@@ -52,9 +52,6 @@ def _resolve_with_default(
 
         has_default = default_value is not DEFAULT_VALUE_MARKER
 
-        if has_default and _get_value(value) is None:
-            return default_value
-
```

Comment:
it looks like the default value is only coming into play on a missing value now.
is it handled somewhere else as well?
why does this work?
```python
assert OmegaConf.create().get("foo", "bar") == "bar"
```

Under the assumption that the above case is handled at a higher level, maybe we should just handle the MISSING case at a higher level and change the function signature to reflect it?

Reply:
> it looks like the default value is only coming into play on a missing value now.

That's correct.

> is it handled somewhere else as well?

Not sure exactly what you mean here.

> why does this work?
> 
> ```python
> assert OmegaConf.create().get("foo", "bar") == "bar"
> ```

Because here we only call `_resolve_with_default()` if the node associated to `foo` exists. 

> Under the assumption that the above case is handled at a higher level, maybe we should just handle the MISSING case at a higher level and change the function signature to reflect it?

I actually considered it -- that would mean removing `_resolve_with_default()` and calling directly `_maybe_resolve_interpolation()` instead. But then there are ~4 places where we would need to implement a similar logic to handle missing values => it made sense to me to keep it there.

Looking more closely at it though, I was wondering about why `value` was of type `Any` instead of `Node`, and it turns out it is possible to make it `Node` (which makes it easier to understand what is going on) => I did that in fa6f68f